crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -359,22 +359,26 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                     .map_or(preceding.end(), |comment| comment.slice().end());
 
                 match node_level {
-                    NodeLevel::TopLevel(_) => match lines_after(end, source) {
-                        0 | 1 => hard_line_break().fmt(f)?,
-                        2 => empty_line().fmt(f)?,
-                        _ => match source_type {
-                            PySourceType::Stub => {
-                                empty_line().fmt(f)?;
-                            }
-                            PySourceType::Python | PySourceType::Ipynb => {
-                                write!(f, [empty_line(), empty_line()])?;
-                            }
-                        },
-                    },
-                    NodeLevel::CompoundStatement => match lines_after(end, source) {
-                        0 | 1 => hard_line_break().fmt(f)?,
-                        _ => empty_line().fmt(f)?,
-                    },
+                    NodeLevel::TopLevel(_) => {
+                        match lines_after_ignoring_end_of_line_trivia(end, source) {
```

Comment:
I'm a bit worried about changing this to `lines_after_ignoring_end_of_line_trivia`. I haven't been able to come up with an example where it is causing troubles but my worry is about that this now counts the empty lines after a trailing comment, but `FormatTrailingComments` sometimes counts the lines up to the previous comment... where it could be possible that we duplicate the empty lines. 

We need to have a closer look how the two play together.